accumulo/src/main/scala/geotrellis/spark/io/accumulo/AccumuloLayerDeleter.scala

```diff
@@ -16,19 +16,21 @@
 
 package geotrellis.spark.io.accumulo
 
-import geotrellis.spark.LayerId
+import geotrellis.spark.{Boundable, LayerId}
 import geotrellis.spark.io._
-
 import com.typesafe.scalalogging.LazyLogging
+import geotrellis.spark.io.avro.AvroRecordCodec
 import org.apache.accumulo.core.client.{BatchWriterConfig, Connector}
 import org.apache.accumulo.core.security.Authorizations
 import org.apache.accumulo.core.data.{Range => AccumuloRange}
+import spray.json.JsonFormat
 
 import scala.collection.JavaConverters._
+import scala.reflect.ClassTag
 
 class AccumuloLayerDeleter(val attributeStore: AttributeStore, connector: Connector) extends LazyLogging with LayerDeleter[LayerId] {
 
-  def delete(id: LayerId): Unit = {
+  def delete[K: ClassTag](id: LayerId): Unit = {
```

Comment:
Was able to reduce the number of required bounds/evidence parameters to just the ClassTag itself... evidently that's all that is required to look up the `KeyIndex`.  

I'm curious as to whether the metadata in the attributes table that's already loaded in the `delete` implementation contains the required classname so that we could materialize the class/type required at runtime rather than have to rely upon an API change.  

Even if that information is available though, I may need an assist using it.  I'm not sure how I'd go from a reflective call to look up a class by name to a type constructor argument required for the `KeyIndex` lookup.